rfc/20241206-oci-registries.md

```diff
@@ -0,0 +1,58 @@
+
+In contrast, running an OpenTofu / Terraform registry requires the setup of [an extra piece of software](https://awesome-opentofu.com/#registry), which incurs additional costs when run publicly and an additional compliance burden when run in an organization.
+
+In addition to the convenience of using existing OCI registry deployments, many registries such as [Harbor](https://goharbor.io/) offer built-in security, license and compliance scanning of container images out of the box, with tools such as [Trivy](https://trivy.dev/) or [Clair](https://clairproject.org/). These security scanners automatically scan container image contents as well as the Go binaries contained in them for possible known vulnerabilities and report them. Furthermore, many of these tools are able to create [Software Bill of Material (SBOM)](https://www.nist.gov/itl/executive-order-14028-improving-nations-cybersecurity/software-security-supply-chains-software-1) artifacts, which is now required by law in many cases. These are features that OpenTofu/Terraform registries do not have today.
+
+## Parts of this RFC
+
+In order to make this RFC easier to read, we have split it into several parts:
+
+1. [A primer on OCI](20241206-oci-registries/1-oci-primer.md)
+2. [Survey results](20241206-oci-registries/2-survey-results.md)
+3. [Design considerations](20241206-oci-registries/3-design-considerations.md)
+4. [Providers in OCI](20241206-oci-registries/4-providers.md)
+5. [Modules in OCI](20241206-oci-registries/5-modules.md)
+6. [Authentication](20241206-oci-registries/6-authentication.md)
+7. [Open questions](20241206-oci-registries/7-open-questions.md)
+
+### Appendices
+
+The following additional sections discuss potential implementation details of the features described in the earlier chapters. These are primarily for core team reference purposes, but potentially interesting to others too.
+
+8. [Authentication-related implementation details](20241206-oci-registries/8-auth-implementation-details.md)
+9. [Provider installation implementation details](20241206-oci-registries/9-provider-implementation-details.md)
+10. [Module installation implementation details](20241206-oci-registries/10-module-implementation-details.md)
+
+## Potential alternatives
+
+OCI Registry support resolves a very real pain-point for enterprise users wanting to run a private registry. A potential alternative would be making the use of a private registry easier, or creating a tool that can maintain a private registry purely based on static files. On that note, we could also implement [running the OpenTofu Registry on the same dataset privately](https://github.com/opentofu/registry/issues/1518).
```

Comment:
Thinking out-loud...

 We could instead build a service that you deploy on your machine or in your infrastructure that implements the registry mirror protocol, but uses an OCI registry as the backing data source.

This has some obvious downsides, like having to run and maintain the services somewhere and managing new or pass-through credentials.

In hindsight, it might have made sense to start with the mirror service approach as a MVP, but IMO we have already identified that users want this integrated into OpenTofu in a reasonably seamless fashion.

Reply:
I will add some additional words here about that possibility, but since each new push to this branch invalidates all previous approvals I'm going to wait a little to do that just to give some time for other potential feedback and then I can push a single commit covering all of the comments and thus hopefully need only one round of re-requesting reviews.